src/System.Memory/tests/Memory/Trim.cs

```diff
@@ -0,0 +1,200 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+namespace System.MemoryTests
+{
+    partial class MemoryTests
+    {
+        [Theory]
+        [InlineData(new int[] { 1 }, 1, new int[] { })]
+        [InlineData(new int[] { 2 }, 1, new int[] { 2 })]
+        [InlineData(new int[] { 1, 2, 1 }, 1, new int[] { 2, 1 })]
+        [InlineData(new int[] { 1, 1, 2, 1 }, 1, new int[] { 2, 1 })]
+        [InlineData(new int[] { 1, 1, 2, 1 }, 2, new int[] { 1, 1, 2, 1 })]
+        [InlineData(new int[] { 1, 1, 2, 1 }, 3, new int[] { 1, 1, 2, 1 })]
+        [InlineData(new int[] { 1, 1, 1, 2 }, 1, new int[] { 2 })]
+        [InlineData(new int[] { 1, 1, 1, 1 }, 1, new int[] { })]
+        public static void MemoryExtensions_TrimStart_Single(int[] values, int trim, int[] expected)
+        {
+            Memory<int> memory = new Memory<int>(values).TrimStart(trim);
+            Assert.True(System.Linq.Enumerable.SequenceEqual(expected, memory.ToArray()));
```

Comment:
Used `int` to avoid any special-case branches for `byte` or `char` (now or in the future)